Source: COMCAST
Upstream-Status: N/A,as the patch is specific to Comcast and Video Accelerators
From: Lakshmipriya <LakshmipriyaLakshmipriya_Purushan@cable.comcast.com>
---

diff --git a/src/lib/rootfs.py b/src/lib/rootfs.py
index db15768..3091e6f 100644
--- a/src/lib/rootfs.py
+++ b/src/lib/rootfs.py
@@ -181,6 +181,12 @@ class cRootfs(object):
         if error:
             raise Exception("program returned error code {%s}"%(error))
 
+        # set exec permission for rootfs directory for capkeep capabilities support
+        linux_cmd = "chmod o+x %s"%self.getPathRootfsHost()
+        error = os.system(linux_cmd)
+        if error:
+            raise Exception("program returned error code {%s}"%(error))
+
         # set proper rights for container launcher
         if (self.launcherName != None and self.launcherName != " "):
             linux_cmd = "chmod 550 %s -R "%self.getLauncherFileHost()

